flashinfer/cute_dsl_attention.py

```
@@ -0,0 +1,283 @@
+        raise ValueError("query, key, and value must be on the same device.")
+    if query.shape[0] != key.shape[0] or query.shape[0] != value.shape[0]:
+        raise ValueError("query, key, and value must have the same batch size.")
+    if key.shape[1] != value.shape[1]:
+        raise ValueError("key and value must have the same sequence length.")
+    if query.shape[2] != key.shape[2]:
+        raise ValueError(
+            "GQA is not yet supported; query and key must have the same "
+            "number of heads."
+        )
+    if key.shape[2] != value.shape[2]:
+        raise ValueError("key and value must have the same number of KV heads.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)